papertty.py

```diff
@@ -475,32 +485,31 @@ def sigusr1_handler(sig, frame):
                 oldbuff = ''
                 flags['scrub_requested'] = False
             with open(vcsa, 'rb') as f:
-                # read the first 4 bytes to get the console attributes
-                attributes = f.read(4)
-                rows, cols, x, y = list(map(ord, struct.unpack('cccc', attributes)))
-
-                # read rest of the console content into buffer
-                buff = f.read()
-                # SKIP all the attribute bytes
-                # (change this (and write more code!) if you want to use the attributes with a
-                # three-color display)
-                buff = buff[0::2]
-                # find character under cursor (in case using a non-fixed width font)
-                char_under_cursor = buff[y * rows + x]
-                cursor = (x, y, char_under_cursor)
-                # add newlines per column count
-                buff = ''.join([r.decode(ptty.encoding, 'ignore') + '\n' for r in ptty.split(buff, cols)])
-                # do something only if content has changed or cursor was moved
-                if buff != oldbuff or cursor != oldcursor:
-                    # show new content
-                    oldimage = ptty.showtext(buff, fill=ptty.black, cursor=cursor if not nocursor else None,
-                                             oldimage=oldimage,
-                                             **textargs)
-                    oldbuff = buff
-                    oldcursor = cursor
-                else:
-                    # delay before next update check
-                    time.sleep(float(sleep))
+                character_width, vcsudev = ptty.vcsudev(vcsa)
```

Comment:
This could be moved to line 468 so that the "Started displaying..." notice shows the actual device being used (`vcsu` or `vcs`) and since there's no actual need to call the function repeatedly after startup.

That is, something like this (this assumes the `vcsudev` is still static and has the `force_vcs` argument):

```python
character_width, vcsudev = ptty.vcsudev(vcsa, force_vcs=not isinstance(ptty.font, ImageFont.FreeTypeFont))
print("Started displaying {}, minimum update interval {} s, exit with Ctrl-C".format(vcsudev, sleep))
```

Reply:
That's a good point, those two variables should be pretty constant within the loop.